presto-main/src/main/java/io/prestosql/memory/MemoryPool.java

```diff
@@ -338,12 +338,14 @@ private synchronized void updateTaggedMemoryAllocations(QueryId queryId, String
         Map<String, Long> allocations = taggedMemoryAllocations.computeIfAbsent(queryId, ignored -> new HashMap<>());
         allocations.compute(allocationTag, (ignored, oldValue) -> {
             if (oldValue == null) {
+                checkArgument(delta > 0, "negative delta (%s) for new tag [%s]", delta, allocationTag);
```

Comment:
what about `FORCE_FREE_TAG` which has negative delta?

Reply:
the delta can negative, but it should not fall into `oldValue == null` case, as whenever we're using FORCE_FREE_TAG, we're zeroing existing allocation

(the delta can be also be 0, this should be handled correctly at point correctly)